server/tests/models/test_causal_lm.py

```diff
@@ -54,38 +53,43 @@ def default_multi_requests_causal_lm_batch(default_pb_request, gpt2_tokenizer):
 
     batch_pb = generate_pb2.Batch(id=1, requests=[req_0, req_1], size=2)
     return CausalLMBatch.from_pb(
-        batch_pb, gpt2_tokenizer, torch.float32, torch.device("cpu")
+        batch_pb, gpt2_tokenizer, torch.float32, torch.device("hpu")
     )
 
 
 def test_batch_from_pb(default_pb_batch, default_causal_lm_batch):
     batch = default_causal_lm_batch
 
     assert batch.batch_id == default_pb_batch.id
-    assert batch.requests == default_pb_batch.requests
+    assert len(batch.requests) == len(default_pb_batch.requests)
+
+    for r in range(0,len(default_pb_batch.requests)):
+        assert batch.requests[r].data == default_pb_batch.requests[r]
+
+    # For Gaudi we are adding padding of multiplication of bucket size
+    size_of_padded_to_bucket = ((default_pb_batch.size + PREFILL_BATCH_BUCKET_SIZE - 1) // PREFILL_BATCH_BUCKET_SIZE) * PREFILL_BATCH_BUCKET_SIZE
+
+    assert len(batch.input_ids) == size_of_padded_to_bucket 
 
-    assert len(batch.input_ids) == default_pb_batch.size
     assert batch.input_ids[0][-1] == 14402
```

Comment:
Is it true? I would expect that the last element in input_ids will be a pad token (place for first output token). From my perspective it should be like below:
```
assert batch.input_ids[0][-1] == 50256
assert batch.input_ids[0][-2] == 14402
assert torch.all(batch.input_ids[0][:-2] == 50256)
```

Reply:
@kdamaszk In this particular test we have our tensor : batch.input_ids of shape [4,1] e.g.:
```bash
(Pdb) p batch.input_ids.size()

torch.Size([4, 1])
```
content of this tensor is a follows:
```bash
(Pdb) p batch.input_ids

tensor([[14402],
        [   30],
        [   30],
        [   30]], device='hpu:0')
```
Then when we inspect batch.input_ids[0] is is of `size 1` :
```bash
(Pdb) p batch.input_ids[0]
tensor([14402], device='hpu:0')

```

Hence ```batch.input_ids[0][-1]``` is:
```bash
(Pdb) p batch.input_ids[0][-1]

tensor(14402, device='hpu:0')
```

Reply:
It means that you fall into the path "is_optimized_for_gaudi=False". At the beginning, Gaudi related changes were added with flag "is_optimized_for_gaudi=True". However, nobody was tested the other path (False) so I decided to remove it. It looks like you need to rebase your branch with habana-main and pull the latest changes.